modules/candidate_list/jsx/candidateListIndex.js

```diff
@@ -0,0 +1,363 @@
+        'show': true,
+        'filter': {
+          name: 'project',
+          type: 'select',
+          options: options.project,
+        },
+      },
+      {
+        'label': 'Subproject',
+        'show': true,
+        'filter': {
+          name: 'subproject',
+          type: 'select',
+          options: options.subproject,
+        },
+      },
+      {
+        label: 'Entity Type',
+        show: true,
+        filter: {
+          name: 'entityType',
+          type: 'select',
+          options: {
+            'Human': 'Human',
+            'Scanner': 'Scanner',
+          },
+        },
+      },
+      {
+        'label': 'Scan Done',
+        'show': true,
+        'filter': {
+          name: 'scanDone',
+          type: 'select',
+          hide: this.state.hideFilter,
+          options: {
+            'Y': 'Yes',
+            'N': 'No',
+          },
+        },
+      },
+      {
+        'label': 'Participant Status',
+        'show': true,
+        'filter': {
+          name: 'participantStatus',
+          type: 'select',
+          hide: this.state.hideFilter,
+          options: options.participantstatus,
+        },
+      },
+      {
+        'label': 'DoB',
+        'show': true,
+        'filter': {
+          name: 'DoB',
+          type: 'date',
+          hide: this.state.hideFilter,
+        },
+      },
+      {
+        label: 'Sex',
+        show: true,
+        filter: {
+          name: 'sex',
+          type: 'select',
+          hide: this.state.hideFilter,
+          options: {
+            'Male': 'Male',
+            'Female': 'Female',
+          },
+        },
+      },
+      {
+        'label': 'VisitCount',
+        'show': true,
+        'filter': {
+          name: 'visitCount',
+          type: 'text',
+          hide: this.state.hideFilter,
+        },
+      },
+      {
+        'label': 'Feedback',
+        'show': true,
+        'filter': {
+          name: 'feedback',
+          type: 'select',
+          hide: this.state.hideFilter,
+          options: {
+            '0': 'None',
+            '1': 'opened',
+            '2': 'answered',
+            '3': 'closed',
+            '4': 'comment',
+          },
+        },
+      },
+      {
+        'label': 'Latest Visit Status',
+        'show': true,
+        'filter': {
+          name: 'latestVisitStatus',
+          type: 'select',
+          hide: this.state.hideFilter,
+          options: {
+            'Not Started': 'Not Started',
+            'Screening': 'Screening',
+            'Visit': 'Visit',
+            'Approval': 'Approval',
+            'Recycling Bin': 'Recycling Bin',
+          },
+        },
+      },
+
+    ];
+
+    if (options.useedc === 'true') {
+      fields.push(
+        {
+          'label': 'EDC',
+          'show': true,
+          'filter': {
+            name: 'edc',
+            type: 'date',
+            hide: this.state.hideFilter,
+          },
+        }
+      );
+    }
```

Comment:
Mutation of the array (i.e. `fields.push`) should generally be avoided in functional programming. But I don't think it's a big deal here.

Reply:
You could either reassign the value of fields by doing something like: 
```
let fields = [...fields,
        {
           'label': 'EDC',
           'show': true,
           'filter': {
             name: 'edc',
             type: 'date',
             hide: this.state.hideFilter,
           },
         }
]
```

Or the alternative would be to use a `map` or `reduce` function. I honestly think in this case, after taking a look at it, using `fields.push` is just fine!

Reply:
good to know. Thanks for the suggestion!